packages/xrpl/src/Wallet/index.ts

```diff
@@ -198,27 +244,30 @@ class Wallet {
   }
 
   /**
-   * Derives a wallet from an entropy (array of random numbers).
+   * Derives a wallet from a RFC1751 mnemonic, which is how `wallet_propose` encodes mnemonics.
    *
-   * @param entropy - An array of random numbers to generate a seed used to derive a wallet.
+   * @param mnemonic - A string consisting of words (whitespace delimited) used to derive a wallet.
    * @param opts - (Optional) Options to derive a Wallet.
-   * @param opts.algorithm - The digital signature algorithm to generate an address for.
    * @param opts.masterAddress - Include if a Wallet uses a Regular Key Pair. It must be the master address of the account.
-   * @returns A Wallet derived from an entropy.
+   * @param opts.algorithm - The digital signature algorithm to generate an address for.
+   * @returns A Wallet derived from a mnemonic.
    */
-  public static fromEntropy(
-    entropy: Uint8Array | number[],
-    opts: { masterAddress?: string; algorithm?: ECDSA } = {},
+  private static fromRFC1751Mnemonic(
+    mnemonic: string,
+    opts: { masterAddress?: string; algorithm?: ECDSA },
   ): Wallet {
-    const algorithm = opts.algorithm ?? DEFAULT_ALGORITHM
-    const options = {
-      entropy: Uint8Array.from(entropy),
-      algorithm,
+    const seed = rfc1751MnemonicToKey(mnemonic)
+    let encodeAlgorithm: 'ed25519' | 'secp256k1'
+    if (opts.algorithm === ECDSA.ed25519) {
+      encodeAlgorithm = 'ed25519'
+    } else {
+      // Defaults to secp256k1 since that's the default for `wallet_propose`
+      encodeAlgorithm = 'secp256k1'
     }
-    const seed = generateSeed(options)
-    return Wallet.deriveWallet(seed, {
-      algorithm,
+    const encodedSeed = encodeSeed(seed, encodeAlgorithm)
```

Comment:
`encodeSeed` should take `ECDSA` too, not a string - then the previous bit becomes 
```
const encodeAlgorithm = opts.algorithm ?? ECDSA.secp256k1
```

Reply:
That'd be a breaking change to the address-codec

Reply:
Whoops, didn't realize this was an address-codec method